### PR TITLE
Add break points in other modules.

### PR DESCRIPTION
**Pull request by DavidEichmann**

### CODE REVIEW

File: src/lib/GHCIDebug.ts [hunk 1]
```diff
@@ -102,7 +102,12 @@ export class GHCIDebug {
     if (typeof breakpoint === 'string') {
       this.run(`:break ${breakpoint}`)
     } else {
-      this.run(`:break ${breakpoint.file} ${breakpoint.line}`)
+      this.run(`:show modules`)
```
Comment (lierdakil) [inline src/lib/GHCIDebug.ts]: Template string doesn't seem necessary here?

File: src/lib/GHCIDebug.ts [hunk 2]
```diff
@@ -102,7 +102,12 @@ export class GHCIDebug {
...
+        .then(modules => {
+          const matchResult = modules.match(new RegExp('([^ ]+) +\\( +' + breakpoint.file))
+          if (!matchResult) { return }
```
Comment (lierdakil) [inline src/lib/GHCIDebug.ts]: Probably a good idea to add a notification that something went wrong. `atom.notifications.addError` seems like a reasonable way to do it.

File: src/lib/GHCIDebug.ts [hunk 3]
```diff
@@ -102,7 +102,12 @@ export class GHCIDebug {
     if (typeof breakpoint === 'string') {
       this.run(`:break ${breakpoint}`)
     } else {
-      this.run(`:break ${breakpoint.file} ${breakpoint.line}`)
+      this.run(`:show modules`)
+        .then(modules => {
```
Comment (lierdakil) [inline src/lib/GHCIDebug.ts]: It boils down to personal preference, but I would prefer `await` instead of `then`.

**Comment by lierdakil** [comment]
Cool, thanks. There are a couple changes I'd like to make, but I could do those myself easily enough.

**Review by lierdakil — COMMENTED** [review]
For posterity.

**Comment by DavidEichmann** [comment]
I've also realized that it would probably be best to cache the results of :show module. Else it will be called for each break point.

**Comment by lierdakil** [comment]
That's a reasonable thought. Well, if you'd like to work on it some more, I'm all for it.

**Comment by lierdakil** [comment]
If you're going to, pull the changes from https://github.com/atom-haskell/haskell-debug/tree/DavidEichmann-master -- besides the stuff I commented on already, there's a bugfix in regexp (it should be multiline and anchor to start-of-line, otherwise it captures newlines, which leads to pain and suffering)

**Comment by DavidEichmann** [comment]
I'll give it a shot

**Comment by DavidEichmann** [comment]
I've got to get going, but I've attempted to cache the module paths. There seems to be an infinite loop, I guess with the regex code. Perhaps you can find out what is going wrong.

**Comment by lierdakil** [comment]
Okay. From a quick look this feels more than a bit overkill though. `typescript-collections` seems extraneous, since we have `Map`, and code would probably be easier to read if you factored getting module name from file path into a new function.

**Pull request merged**